crates/exex/exex/src/manager.rs

```diff
@@ -109,7 +109,9 @@ impl ExExHandle {
                 // [ExExNotification::ChainReverted] cases and always send the
                 // notification, because the ExEx should be aware of the reorgs and reverts lower
                 // than its finished height
-                ExExNotification::ChainReorged { .. } | ExExNotification::ChainReverted { .. } => {}
+                ExExNotification::ChainReorged { .. } |
+                ExExNotification::ChainReverted { .. } |
+                ExExNotification::BlockFinalized(_) => {}
```

Comment:
hmm, I think this variant should be in the arm above where we handle `ExExNotification::ChainCommitted`. @onbjerg wdyt?

Reply:
I am thinking here out loud that would there be a case where we receive a notification about a finalized block with lower height than an already finalized block?